custom-metrics-stackdriver-adapter/pkg/cmd/server/start.go

```diff
@@ -73,15 +73,15 @@ func (o CustomMetricsAdapterServerOptions) Config() (*apiserver.Config, error) {
 		return nil, fmt.Errorf("error creating self-signed certificates: %v", err)
 	}
 
-	serverConfig := genericapiserver.NewConfig(apiserver.Codecs)
-	if err := o.SecureServing.ApplyTo(serverConfig); err != nil {
+	serverConfig := genericapiserver.NewRecommendedConfig(apiserver.Codecs)
+	if err := o.SecureServing.ApplyTo(&serverConfig.Config); err != nil {
```

Comment:
There is also `NewRecommendOptions` with one "big" `ApplyTo`.  That way you won't forget to initialize important fields.

Reply:
compare https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go#L40.

Reply:
Done